aws_lambda_builders/workflows/rust_cargo/actions.py

```diff
@@ -0,0 +1,228 @@
+"""
+Rust Cargo build actions
+"""
+
+import os
+import subprocess
+import shutil
+import json
+
+from aws_lambda_builders.workflow import BuildMode
+from aws_lambda_builders.actions import ActionFailedError, BaseAction, Purpose
+
+
+class BuilderError(Exception):
+    MESSAGE = "Builder Failed: {message}"
+
+    def __init__(self, **kwargs):
+        Exception.__init__(self, self.MESSAGE.format(**kwargs))
+
+
+class OSUtils(object):
+    """
+    Wrapper around file system functions, to make it easy to
+    unit test actions in memory
+    """
+
+    def popen(self, command, stdout=None, stderr=None, env=None, cwd=None):
+        return subprocess.Popen(command, stdout=stdout, stderr=stderr, env=env, cwd=cwd)
+
+    def copyfile(self, source, destination):
+        shutil.copyfile(source, destination)
+
+    def makedirs(self, path):
+        if not os.path.exists(path):
+            os.makedirs(path)
+
+
+def parse_handler(handler):
```

Comment:
I can. Have unit test coverage for the current implementation but I'm less familiar with python idioms so please advise where I'm way of the rails on what's normal

Reply:
actually taking a closer look `rsplit` is not what we want because that splits on the right. we follow the bin package model specifier spec'd in the design doc. we actually want the first segment to represent the cargo package and the remainder the bin name. In your example `a.b` is not a valid cargo package name.

Rust does not have deeply nested package identifiers like you might find with java. A few example projects are included in tests. 

No change is needed here.